src/diffusers/models/unet_2d_condition.py

```diff
@@ -689,22 +689,34 @@ def forward(
         sample = self.conv_in(sample)
 
         # 3. down
+
+        is_controlnet = mid_block_additional_residual is not None and down_block_additional_residuals is not None
+        is_adapter = mid_block_additional_residual is None and down_block_additional_residuals is not None
```

Comment:
@patrickvonplaten This is from the starter code you had recommended, could we instead pass a flag to forward indicating if the additional residuals should be treated as controlnet or adapter residuals?

I.e.

`forward(additional_residual_type: Optional[str] = "controlnet")` 

Inferring which it is from the arguments is nice because it saves another argument to the forward method but is potentially error prone in the future if we need to re-use the residual arguments

Reply:
Not a fan of adding flag arguments to the forward, ideally we should only have tensors there. 
For now I'd like to leave as is. We can always add a flag in the future if we believe it's necesasry